core/platform/taskqueue/cloud_taskqueue_services.py

```diff
@@ -30,14 +31,21 @@
 from google.cloud import tasks_v2
 from google.protobuf import timestamp_pb2
 
+from typing import Any, Dict, Optional # isort:skip # pylint: disable=unused-import
+
 CLIENT = tasks_v2.CloudTasksClient(
     credentials=(
         auth.credentials.AnonymousCredentials()
         if constants.EMULATOR_MODE else auth.default()[0]))
 
 
 def create_http_task(
-        queue_name, url, payload=None, scheduled_for=None, task_name=None):
+        queue_name: str,
+        url: str,
+        payload: Optional[Dict[str, Any]] = None,
```

Comment:
Missing TODO/Justification of why this usage of `Dict[str, Any]` is valid.

Reply:
Moved to doc

Reply:
I think this usage is justified since it's mentioned in docstring (sorry didn't pay attention to it earlier) as well as explained via examples in the doc (see the [bookmark](https://docs.google.com/document/d/1LEWKK2H303eHmWg1hHN2EMIlnewB6EnEyrQoReY-M6s/edit#bookmark=id.x165l86yr83y) to it). Similar for other places in PR also where payload is used and type is already mentioned in docstring.

For places that create a mock function for this, you can have a comment on top explaining the reason - see my other comment here.

Reply:
Although the docstring states some kind of usage, should we also look into our usage of this function, and see if we can limit this arg to some type.

Reply:
Agreed.

Reply:
One more thing, why is this `Optional`?

Reply:
We explicitly check for 'None' in this method. See [here](https://github.com/oppia/oppia/blob/a4a3c2b7c673e928264919b595f51222ddd8e2c5/core/platform/taskqueue/cloud_taskqueue_services.py#L72). If value is not None, we do something. Hence I think optional should be used.

Reply:
Makes sense.